test/unit/callbacks_and_delays_test.lua

```diff
@@ -0,0 +1,86 @@
+local fiber = require("fiber")
+local t = require("luatest")
+
+local helpers = require("test.helper")
+
+local g = t.group('callbacks_and_delays', {
+    {index_type = 'TREE', engine = 'vinyl'},
+    {index_type = 'TREE', engine = 'memtx'},
+    {index_type = 'HASH', engine = 'memtx'},
+})
+
+g.before_each({index_type = 'TREE'}, function(cg)
+    t.skip_if(cg.params.engine == 'vinyl' and helpers.vinyl_is_broken(),
+            'Blocked by https://github.com/tarantool/tarantool/issues/6448')
+    g.space = helpers.create_space_with_tree_index(cg.params.engine)
+end)
+
+g.before_each({index_type = 'HASH'}, function(cg)
+    g.space = helpers.create_space_with_hash_index(cg.params.engine)
+end)
+
+g.before_each({index_type = 'BITSET'}, function(cg)
+    g.space = helpers.create_space_with_bitset_index(cg.params.engine)
+end)
+
+g.after_each(function(g)
+    g.space:drop()
+end)
+
+function g.test_delays_and_scan_callbacks(cg)
+    local space = cg.space
+    local total = 10
+    for i = 1, total do
+        space:insert({i, tostring(i)})
+    end
+
+    -- To check all delays (iteration and full scan), two full scan
+    -- iterations will be performed.
+    local first_fullscan_done = false
+    local cond = fiber.cond()
+    local start_time = 0
+    local complete_time = 0
+
+    local iteration_delay = 1
+    local full_scan_delay = 2
+    local full_scan_success_counter = 0
+
+    local task = expirationd.start("test", space.id, helpers.is_expired_true,
+            {
+                iteration_delay = iteration_delay,
+                full_scan_delay = full_scan_delay,
+                tuples_per_iteration = 5,
+                on_full_scan_start = function(_)
+                    start_time = fiber.time()
+                    if first_fullscan_done then
+                        -- Check the full scan delay with an accuracy of 0.1 seconds.
+                        -- Difference between start time of the second full scan
+                        -- and complete_time of the first full scan.
+                        t.assert_almost_equals(start_time - complete_time, full_scan_delay, 0.1)
+                    end
+                end,
+                on_full_scan_success = function(_)
+                    -- Must be called twice.
+                    full_scan_success_counter = full_scan_success_counter + 1
+                end,
+                on_full_scan_complete = function(_)
+                    complete_time = fiber.time()
+                    if first_fullscan_done then
+                        cond:signal()
+                    else
+                        -- Check the accuracy of iteration delay.
+                        -- Difference between start time and complete_time of the first full scan.
+                        t.assert_almost_equals(complete_time - start_time, iteration_delay, 0.1)
+                        first_fullscan_done = true
+                    end
+                end
+            }
+    )
```

Comment:
I propose to create a named functions and pass them by name to `expirationd.start()`.

Reply:
Fixed